sources/web/datalab/server.ts

```diff
@@ -141,6 +152,11 @@ function handleRequest(request: http.ServerRequest,
       (path.indexOf('/files') == 0) ||
       (path.indexOf('/edit') == 0) ||
       (path.indexOf('/sessions') == 0)) {
+
+    if (path.indexOf('/tree') == 0) {
+        loadedSettings[startup_path_setting] = path
+        settings_.updateUserSetting(userId, startup_path_setting, path);
```

Comment:
updateUserSetting() does synchronous file writing so it is more expensive compared to other requests. Although each Datalab instance serves one user, I would avoid expensive synchronous operations per normal request ('/tree'), since it will block other requests.

Perhaps at least we should make the file writing asynchronous. Ideally there is a way to detect VM shutdown so we can write the user config to disk only once.